polkadot/xcm/pallet-xcm-benchmarks/src/generic/benchmarking.rs

```diff
@@ -77,11 +77,13 @@ benchmarks! {
 		let mut executor = new_executor::<T>(Default::default());
 		executor.set_holding(holding);
 
-		let fee_asset = AssetId(Here.into());
+		// The worst case we want for buy execution in terms of
+		// fee asset and weight
+		let (fee_asset, weight_limit) = T::worst_case_buy_execution()?;
```

Comment:
I definitely like the idea of benchmarking with `WeightLimit::Limited` so we can hit more logic. There's now a `fee_asset` configuration item that is used to buying execution in the benchmark for `refund_surplus`, this could be merged with that to include the weight_limit.

We also have to make sure this is in holding, don't know if there's a good way to do it.

Reply:
ok I can try to do this!

Reply:
For holding, yes we rely on `worst_case_holding` having the fees, and I guess that if it is not the benchmark will fail. So I think this is good enough.

The other way we could do it is by injecting the fee_asset into holding directly, without using `worst_case_holding` at all

Reply:
@franciscoaguirre I dont see the `fee_asset` setup in `refund_surplus`. all I see is that the `total_surplus` is artificially set https://github.com/paritytech/polkadot-sdk/blob/c01dbebeaa6394691974de46dd2d41a582f6a4c2/polkadot/xcm/pallet-xcm-benchmarks/src/generic/benchmarking.rs#L131